### ISSUE TITLE
Cannot Handle Invalid SWF File Error

### ISSUE THREAD

**Issue by MatteoCampinoti94** [ruffle-rs/ruffle#6665]
I've been adding ruffle to my web app, and I'm having trouble handling some edge cases. Specifically, it seems there is no way to catch (and thus handle) errors that arise _after_ the file has been loaded, for example if the file contains errors or is invalid.

These errors are caught correctly by `ruffle_web.js` and logged, but are not allowed to rise and the only hint that something occurred is in the console, no message appears in the player either.

Ideally, these errors would trigger an `error` event in the player object, that way they can be caught and handled. In my case I'm using a small spinner animation, and a hidden container for the ruffle player, and when the `loadeddata` event is triggered, it hides the spinner and unhides the player, but because after load errors are not allowed to rise, there is no way to tell the user that something happened.

**Comment by MatteoCampinoti94** [comment]
This is the "solution" I found for now:

```javascript
const ruffleError = (e) => {
    if (e !== undefined) console.error(e)
    document.getElementById("flash-loading").hidden = true
    document.getElementById("flash-container").hidden = true
    document.getElementById("flash-error").hidden = false
}

const _console_error = console.error;
console.error = function (...arguments) {
    if (String(arguments[0]).toLowerCase() === "asynchronous error occurred: invalid swf") ruffleError();
    return _console_error.apply(console, arguments);
};
```

It works without problems, but unless I add every single error message that `ruffle_web.js` logs, it won't catch everything, and even then it's really a stopgap solution.

**Comment by MatteoCampinoti94** [comment]
Maybe it would be possible to pass event callbacks when creating the player instead of having these events rise?

For example:

```javascript
player.addEventCallback("invalidswf", doSomething);
```
